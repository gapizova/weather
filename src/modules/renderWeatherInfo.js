const YANDEX_API_KEY = '4f68df8e-cab0-4360-8ebb-d6a2b6c2248c';

/**
 *
 * @param el - an element is passed in which all components will be paint
 * @param data - weather data received in the module getWeather
 */
export function renderWeatherInfo(el, data) {
  const mapUrl = `https://static-maps.yandex.ru/v1?apikey=${YANDEX_API_KEY}&l=map&ll=${data.coord.lon},${data.coord.lat}&z=10`;

  el.innerHTML = `
    <h2>${data.name}</h2>
    <p class="text-weather">${data.weather[0].main} - ${data.weather[0].description}</p>
    <img class="icon-weather" src="https://openweathermap.org/img/wn/${data.weather[0].icon}@2x.png"></img>
    <p class="text-temp">Температура: ${data.main.temp} °C</p>
    <img class="map-weather" src="${mapUrl}" alt="Yandex map"></img>
  `;
}
