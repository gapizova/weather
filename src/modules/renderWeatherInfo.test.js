import { renderWeatherInfo } from './renderWeatherInfo.js';

describe('testing function renderWeatherInfo', () => {
  let el;

  beforeEach(() => {
    el = document.createElement('div');
  });

  it('should render weather information correctly', () => {
    const data = {
      name: 'London',
      weather: [
        {
          main: 'Cloudy',
          description: 'Overcast clouds',
          icon: '04d',
        },
      ],
      main: {
        temp: 15,
      },
      coord: {
        lon: 0,
        lat: 0,
      },
    };

    renderWeatherInfo(el, data);

    expect(el.innerHTML).toMatchSnapshot();
  });

  it('should set the correct src attribute for the weather icon image', () => {
    const data = {
      name: 'London',
      weather: [
        {
          main: 'Sunny',
          description: 'Clear sky',
          icon: '01d',
        },
      ],
      main: {
        temp: 25,
      },
      coord: {
        lon: 0,
        lat: 0,
      },
    };

    renderWeatherInfo(el, data);

    const iconImage = el.querySelector('.icon-weather');
    expect(iconImage.src).toContain(data.weather[0].icon);
  });

  it('should set the correct src attribute for the map image', () => {
    const YANDEX_API_KEY = '4f68df8e-cab0-4360-8ebb-d6a2b6c2248c';

    const data = {
      name: 'London',
      weather: [
        {
          main: 'Sunny',
          description: 'Clear sky',
          icon: '01d',
        },
      ],
      main: {
        temp: 25,
      },
      coord: {
        lon: 0,
        lat: 0,
      },
    };

    renderWeatherInfo(el, data);

    const expectedUrl =
      `https://static-maps.yandex.ru/v1?apikey=${YANDEX_API_KEY}&l=map&ll=` +
      `${data.coord.lon},${data.coord.lat}&z=10`;

    const mapImage = el.querySelector('.map-weather');
    expect(mapImage.src).toBe(expectedUrl);
  });
});
