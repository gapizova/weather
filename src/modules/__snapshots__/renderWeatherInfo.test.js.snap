// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing function renderWeatherInfo should render weather information correctly 1`] = `
"
    <h2>London</h2>
    <p class="text-weather">Cloudy - Overcast clouds</p>
    <img class="icon-weather" src="https://openweathermap.org/img/wn/04d@2x.png">
    <p class="text-temp">Температура: 15 °C</p>
    <img class="map-weather" src="https://static-maps.yandex.ru/v1?apikey=4f68df8e-cab0-4360-8ebb-d6a2b6c2248c&amp;l=map&amp;ll=0,0&amp;z=10" alt="Yandex map">
  "
`;
